### PR TITLE
Feature error handling by git rebase --abort

### LINKED ISSUE #28: Error handling by hard reset
In the current, qs command cannot recover git-rebase context when some conflicts.
In this situation, the qs command should leave to the user's manual operation.

### PR DESCRIPTION
**Pull request by kamontia**
ref: #28 

I have implemented the feature, but I could not prepare to evaluate this situation.

Could you have any idea how to test this feature?

If merged, close #28 .

### CODE REVIEW

**Comment by kamontia** [comment]
@chaspy Hi, please review.

You need to check in your local environment.
Run `test-PID-qs-8`, an error occurs.
```
+ REFLOG_HASH_1=c142450  <- this latest commit hash

...... many logs


Executing: git commit --amend -m"Revert "Add file-8""
error: pathspec 'file-8' did not match any file(s) known to git.
warning: execution failed: git commit --amend -m"Revert "Add file-8""
You can fix the problem, and then run

  git rebase --continue


ERRO[0000] QS reseted this operation. Please git rebase manually 
ERRO[0000] exit status 1  
```

and move to `test-$PID-8` directory, 
```
commit c142450d38a2b9bcb9b9e3d469127f4c2c9d40db (HEAD -> master)   <-  This commit hash equals REFLOG_HASH_1!
Author: git fixup <git-fixup@example.com>
Date:   Fri Aug 24 13:39:32 2018 +0900

    Add file-11

```
and

confirm that Git is not on rebase context.
```
$ git rebase --continue
No rebase in progress?
```

**Comment by chaspy** [comment]
@kamontia 
Still have problems? In my environment, the editor opens when reverted.

File: script/test.sh [hunk 1]
```diff
@@ -61,6 +61,29 @@ test_check () {
...
+   git add file-11
+   git commit -m "Add file-11"
+
+   set +e
+   REFLOG_HASH_1=$(git log --oneline --format=%h|head -n1)
+   ./${ExecComamnd}  -n 2..5 -f -d
+   REFLOG_HASH_2=$(git log --oneline --format=%h|head -n 1)
+   set -e
+   if [ "${REFLOG_HASH_1}" == "${REFLOG_HASH_2}" ]; then
+     echo "[passed] RUN ./${ExecComamnd}  -n 2..5 -f -d"
```
Comment (chaspy) [inline script/test.sh]: please please write to text summary

> >> ./../test-$$-result

L81 too.

**Comment by kamontia** [comment]
@chaspy 
> In my environment, the editor opens when reverted.
This behavior is right.

It means that the test method is NOT good.
In CI server test, seem for me to fail revert commit.

File: script/test.sh (hunk 1 above)
Comment (kamontia) [reply script/test.sh]: Added it.

**Comment by kamontia** [comment]
All problem is solved!!!

**Comment by chaspy** [comment]
thank you ❤️

**Pull request merged**